src/core/maprenderer/qgsmaprendererjob.h

```diff
@@ -352,6 +352,22 @@ class CORE_EXPORT QgsMapRendererJob : public QObject SIP_ABSTRACT
      */
     void setCache( QgsMapRendererCache *cache );
 
+    /**
+     * Returns the label sink associated to this rendering job.
+     * \note not available in Python bindings
+     * \since QGIS 3.24
+     */
+    QgsLabelSink *labelSink() const { return mLabelSink; } SIP_SKIP
+
+    /**
+     * Assigns the label sink which will take over responsability for handling labels
+     * during the rendering job.
+     * \note the ownership is not transferred
```

Comment:
Ownership is not transferred and the sink must exist for the lifetime of the map rendering job.